mycroft/client/speech/listener.py

```diff
@@ -170,137 +65,135 @@ class AudioConsumer(threading.Thread):
     # In seconds, the minimum audio size to be sent to remote STT
     MIN_AUDIO_SIZE = 1.0
 
-    def __init__(
-            self, state, queue, emitter, wakeup_recognizer,
-            wakeword_recognizer, wrapped_remote_recognizer, wakeup_prefixes,
-            wakeup_words):
+    def __init__(self, state, queue, emitter, wakeup_recognizer,
+                 mycroft_recognizer, remote_recognizer):
         threading.Thread.__init__(self)
         self.daemon = True
         self.queue = queue
         self.state = state
         self.emitter = emitter
         self.wakeup_recognizer = wakeup_recognizer
-        self.ww_recognizer = wakeword_recognizer
-        self.wrapped_remote_recognizer = wrapped_remote_recognizer
-        self.wakeup_prefixes = wakeup_prefixes
-        self.wakeup_words = wakeup_words
+        self.mycroft_recognizer = mycroft_recognizer
+        self.remote_recognizer = remote_recognizer
         self.metrics = MetricsAggregator()
 
     def run(self):
         while self.state.running:
-            self.try_consume_audio()
+            self.read_audio()
 
     @staticmethod
     def _audio_length(audio):
-        return float(
-            len(audio.frame_data))/(audio.sample_rate*audio.sample_width)
+        return float(len(audio.frame_data)) / (
+            audio.sample_rate * audio.sample_width)
 
-    def try_consume_audio(self):
+    def read_audio(self):
         timer = Stopwatch()
-        hyp = None
         audio = self.queue.get()
-        self.metrics.timer(
-            "mycroft.recognizer.audio.length_s", self._audio_length(audio))
+        self.metrics.timer("mycroft.recognizer.audio.length_s",
+                           self._audio_length(audio))
         self.queue.task_done()
         timer.start()
+
         if self.state.sleeping:
-            hyp = self.wakeup_recognizer.transcribe(
-                audio.get_wav_data(), metrics=self.metrics)
-            if hyp and hyp.hypstr:
-                logger.debug("sleeping recognition: " + hyp.hypstr)
-            if hyp and hyp.hypstr.lower().find("wake up") >= 0:
-                SessionManager.touch()
-                self.state.sleeping = False
-                self.__speak("I'm awake.")  # TODO: Localization
-                self.metrics.increment("mycroft.wakeup")
+            self.wake_up(audio)
```

Comment:
i think this method name can be more helpful; perhaps self.check_wake_up(audio), as self.wake_up phrased as a verb  suggests the state is being changed unconditionally (as opposed to on the condition that the input matches).